### ISSUE TITLE
Syncing Asus Aura's lighting to all other RGB devices!

### ISSUE THREAD

**Issue by Zenairo** [DarthAffe/RGB.NET#3]
Hello! First off a huge THANK YOU for this awesome project! Your code is extremely well organized and brings some sensibility to the rather awful SDK's provided by these companies. I have learned an enormous amount going through it and creating my own project with it.

I have forked your repo and added a project called AuraSync (unsure how to follow naming conventions properly, I tried my best). My goal here is to take the colors sent to the Aura lighting zones by the actual Aura software and send them to corresponding zones in my Corsair products. This allows Aura to control both Aura and Cue devices. Eventually when Corsair releases Sync It I am hoping the Cue SDK expands to control RGB's on the Lighting Node Pro.

I had to tweak your Aura devices so the motherboard colors could be retrieved then sent to the keyboard and mouse pad. Please let me know what you think! Also, I had some trouble using the conical gradient brush in my rectangles (unsure how to do this?) and included what I had written but commented out.

Eventually I want to create one program that unifies all of my lighting systems (Aura, Cue, Link, Zotac Firestorm) by allowing the user to choose an ecosystem to get colors from and send to all other ecosystems. (So Aura can control Cue/Link/Sync It/Firestorm or so Cue can control Aura/Link/Sync It/Firestorm)

Problems I see are:

Zotac Firestorm does not have an SDK, completely unsure how to control this other than some hacky stuff I don't quite comprehend. Ways around this I can think of are physically connecting my GPU lights to my Aura RBG headers but this isn't a software solution and I want everyone to be able to use this creation without modification. Likely going to have to wait for an SDK or learn how to control it myself. I cannot find any community projects around Zotac Firestorm / Spectra or the protocol to control the lighting on my 1080 Ti's.

The Lighting Node PRO that Link uses to control Corsair RGB strips and fans like the HD120 does not have an SDK  yet, a possible solution here is to wire everything to an Arduino as in http://forum.corsair.com/v3/showthread.php?t=165029 but again this is not a software solution and I want this to all be plug and play. Fortunately it seems that when Corsair releases Sync It "sometime next year" (source: http://forum.corsair.com/v3/showthread.php?p=925470#post925470) the Cue SDK may be expanded to include their Lighting Node Pro. Going to ignore the LNP for now.

Mostly doing this to make all my lighting respond to sound so changes must happen instantly, this means the Lighting Node PRO's current community created interface(http://forum.corsair.com/forums/showthread.php?t=120092) likely isn't fast enough(I could be wrong here), probably a similar case with my Zotac Firestorm lighting, even if I were to figure out the protocol the device likely updates on a delay (just my observation from using Link or Firestorm when applying changes the LNP and Zotac cards take a moment to update).

I look forward to hearing your thoughts on this project! Once again, thank you for this awesome project! Also, if I mention my addition to your work anywhere how should I cite you?

Btw couldn't find a way to contact you other than submitting an issue, apologies if this isn't proper.

Edit:
Hardware: https://pcpartpicker.com/b/9ZcYcf

Using the K95 RGB, MM800, and an Asus Maximus IX Extreme with aura headers BackIO, PCH, Header 1, Header 2.

**Comment by DarthAffe** [comment]
Hey, great to hear that it was of some use for you :)

Just from looking over your code I would say that you're using the library "wrong". (Please don't see that as an offense, read wrong as "not as I intended it to be used" - I know that the lack of any documentation right now makes it hard to follow the ideas built into it.)
That's why the brush and the update-methode doesn't work as expected.
I'll get more into detail here once I get home.

About the problems you see:
Yeah, that's the main issue. Lots of devices don't have SDKs and even if there is one, all of them works different and support different things. That makes it quite hard to work with multiple devices from different vendors. 
To be honest, I'd never recommend anyone to mess with the low-level stuff (usb-packages, memory-hacks for existing software or stuff like that) for something that's not strict for your own use. Even the existing SDKs are creating such a huge amount of problems in different environments (problems with debugging devices I don't own are one of the main reasons why the development if RGB.NET is so slow) - I'm quite sure that the result of working out a usb-protocol just to have it broken after the next update will nearly never worth the effort.

For questions, support and device-testing help for this library or some RGB-device-developer-talk [I've created the RGB.NET discord channel](https://discord.gg/9kytURv).

**Comment by tmushtare** [comment]
Hi there Zenairo, I was wondering if hooking my Evga 2070 rgb header to the mobo header would allow me to use aura sync to control the gpu's rgb lighting. My cell is 770-367-2191 if it would be easier to talk. 
And DarthAffe I would be using your rgbsyncplus but i cannot figure out how to install the program lol

**Comment by Zenairo** [comment]
Just make sure it matches the format of the Aura header and has resistors
or you will burn out leds. I don't have that card.

**Comment by tmushtare** [comment]
So I definitely lied my GPU has nothing but a 6pin and 8pin port for power.
So I have no idea how I could control the rgb on my GPU through my Mobo
software

**Comment by tmushtare** [comment]
Well i found the rgb header on my 2070 but before i attempt something like this, would someone like to explain step by step how to install or download this RGBSyncPlus program?

**Comment by tmushtare** [comment]
Zenairo, Thank you so much for the reply! I found the rgb header on my 2070 but before i attempt something like this, would you please explain step by step what i am supposed to do with the code so i can use rgbsyncplus or use your code? Would love to sync my gpu using a software mod vs hardware mod.
